sound/soc/sof/trace.c

```diff
@@ -241,8 +241,10 @@ static bool sof_trace_set_host_offset(struct snd_sof_dev *sdev, u32 new_offset)
 		/* This is a bit paranoid and unlikely that it is needed */
 		u32 ret = cmpxchg(&sdev->host_offset, host_offset, new_offset);
 
-		if (ret == host_offset)
+		if (ret == host_offset) {
+			dev_dbg(sdev->dev, "trace: new host_offset: %#x\n", new_offset);
```

Comment:
If the debug is enabled we are printing the IPC messages (two per message). With this change it will look like this:
```
ipc rx: 0x90020000: GLB_TRACE_MSG: DMA_POSITION
trace: new host_offset: 0x7920
ipc rx done: 0x90020000: GLB_TRACE_MSG: DMA_POSITION
```
Instead of an empty indication that the position is updated we will know the value as well.
Then we have another print when the host_offset is reset to 0 (either because of closing the trace file or restart), but it is still one additional message.

It is not going to fill the console, but I always wondered what is the content behind the dtrace position update.

I can drop this print if you think we should not have it.

Reply:
I will create a draft PR on top of this to print the trace file reads as well in hope that we can catch the empty trace with that and have some clue on what is going on.